core/modules/layout/layout.module

```diff
@@ -1525,3 +1528,48 @@ function layout_set_handler_contexts(array &$contexts, array $all_contexts, arra
   // All required contexts populated successfully.
   return TRUE;
 }
+
+/**
+ * Implements hook_block_info().
+ */
+function layout_block_info() {
+  $blocks = array();
+  $blocks['custom_text_block'] = array(
+    'info' => t('Custom text'),
+    'description' => t('Create a custom text block'),
+  );
+
+  return $blocks;
+}
+
+
+/**
+ * Implements hook_block_view().
+ */
+function layout_block_view($delta = '', $settings = array(), $contexts = array()) {
+  $block['subject'] = $settings['subject'];
+  $block['content'] = theme('layout_custom_text_block', array('element' => filter_xss(token_replace($settings['content']['value']))));
```

Comment:
This should use `check_markup()` instead of `filter_xss()` and pass in the saved text format.